site-src/geps/gep-718.md

```diff
@@ -0,0 +1,338 @@
+index 458145f..de720cd 100644
+--- a/apis/v1alpha1/shared_types.go
++++ b/apis/v1alpha1/shared_types.go
+@@ -94,51 +94,39 @@ type GatewayReference struct {
+ 	Namespace string `json:"namespace"`
+ }
+ 
+-// RouteForwardTo defines how a Route should forward a request.
+-type RouteForwardTo struct {
+-	// ServiceName refers to the name of the Service to forward matched requests
+-	// to. When specified, this takes the place of BackendRef. If both
+-	// BackendRef and ServiceName are specified, ServiceName will be given
+-	// precedence.
++// Backend defines how and where a request should be forwarded from the Gateway.
++type Backend struct {
++	// Name is the name of the backend resource to forward matched requests to.
+ 	//
+ 	// If the referent cannot be found, the rule is not included in the route.
+ 	// The controller should raise the "ResolvedRefs" condition on the Gateway
+ 	// with the "DegradedRoutes" reason. The gateway status for this route should
+ 	// be updated with a condition that describes the error more specifically.
+ 	//
+-	// The protocol to use is defined using AppProtocol field (introduced in
+-	// Kubernetes 1.18) in the Service resource. In the absence of the
+-	// AppProtocol field a `networking.x-k8s.io/app-protocol` annotation on the
+-	// BackendPolicy resource may be used to define the protocol. If the
+-	// AppProtocol field is available, this annotation should not be used. The
+-	// AppProtocol field, when populated, takes precedence over the annotation
+-	// in the BackendPolicy resource. For custom backends, it is encouraged to
+-	// add a semantically-equivalent field in the Custom Resource Definition.
++	// +kubebuilder:validation:MinLength=1
++	// +kubebuilder:validation:MaxLength=253
++	Name string `json:"name"`
++
++	// Kind is kind of the backend resource.
+ 	//
+-	// Support: Core
++	// Support: Core (Kubernetes Service)
++	// Support: Custom (any other resource)
+ 	//
+ 	// +optional
++	// +kubebuilder:validation:MinLength=1
+ 	// +kubebuilder:validation:MaxLength=253
+-	ServiceName *string `json:"serviceName,omitempty"`
++	// +kubebuilder:default="service"
++	Kind *string `json:"kind"`
+ 
+-	// BackendRef is a reference to a backend to forward matched requests to. If
+-	// both BackendRef and ServiceName are specified, ServiceName will be given
+-	// precedence.
+-	//
+-	// If the referent cannot be found, the rule is not included in the route.
+-	// The controller should raise the "ResolvedRefs" condition on the Gateway
+-	// with the "DegradedRoutes" reason. The gateway status for this route should
+-	// be updated with a condition that describes the error more specifically.
++	// Group is the group of the backend resource.
+ 	//
+-	// Support: Custom
+-	//
+-	// +optional
+-	BackendRef *LocalObjectReference `json:"backendRef,omitempty"`
++	// +kubebuilder:validation:MinLength=1
++	// +kubebuilder:validation:MaxLength=253
++	Group string `json:"group"`
```

Comment:
Nit: I'm more used to Group>Kind>Name ordering to more closely match the hierarchy.

Reply:
I'll take care of this in the actual PR. Please poke again if you feel strongly.